### DISCUSSION TITLE
Cannot Connect via IP directly

### DISCUSSION THREAD

**Discussion by Creling** [rustdesk/rustdesk#7499] [General]
### Bug Description

While attempting to connect to another PC (PC B) via its IP address, RustDesk displays an error message: `Failed to connect to 100.109.187.106:21118:`. Here, the IP address `100.109.187.106` corresponds to PC B.

Upon checking the listening ports of rustdesk.exe, I discovered that the rustdesk.exe on PC B only listens on port 21119, not 21118.

What can I do to connect PC B via its ip address?

### How to Reproduce

N/A

### Expected Behavior

N/A

### Operating system(s) on local side and remote side

Window 11 -> Windows 10

### RustDesk Version(s) on local side and remote side

1.2.4 -> 1.2.4

### Screenshots

![image](https://github.com/rustdesk/rustdesk/assets/43109504/be46a12d-ee70-4517-a435-ad48efd11666)

**Comment by Wilson16888** [comment]
The environment is same as yours (win11 -> win10 ) , try to re-install the programme under a different path and it can connect normally, but encounter the same problem after reboot device (win11)

**Comment by DragoDeLune** [comment]
I have the same error but in my case both machine run on Win 11. i also notice it append when the lock screen is on
i can use teamviewer to connect to the remove and then i was able to re-use rustdesk i'm using direct ip connection (no server)

**Comment by naides** [comment]
I have encountered the same issue. I changed the port to a random high number, 31128. Although I have to specify the port in the address (e.g., 192.168.50.34:31128), the connection establishes seamlessly. I have observed this behavior on both Linux and Windows hosts.

**Reply by codekain** [reply]
Indeed. changed the direct IP connection port and it worked right away. so happy i read this and didn't go down the rabbit hole of firewall rules and such.

**Comment by callumsmithau** [comment]
It's possible direct IP access and the suggested port 21118 is not actually enabled/commited until you click the adjacent apply button, hence the not listening and why changing port works
